bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerOpenOp.java

```diff
@@ -64,25 +66,39 @@
      * @param cb
      * @param ctx
      */
+    // needed strictly for cases when BK still has ledgers created by versions < 4.2
     public LedgerOpenOp(BookKeeper bk, long ledgerId, DigestType digestType, byte[] passwd,
                         OpenCallback cb, Object ctx) {
         this.bk = bk;
         this.ledgerId = ledgerId;
         this.passwd = passwd;
         this.cb = cb;
         this.ctx = ctx;
-        this.digestType = digestType;
+        this.enableDigestAutodetection = bk.conf.getEnableDigestTypeAutodetection();
+        this.suggestedDigestType = digestType;
     }
 
+    public LedgerOpenOp(BookKeeper bk, long ledgerId, byte[] passwd,
```

Comment:
@eolivelli Done. Initially I started with deprecation of constructor that requires digestType and added this one instead but this leads to maven errors (deprecated method use warning is treated as error). removal of that constructor breaks backwards compat so I decided not to follow through with it.